### DISCUSSION TITLE
RTC on Orange Pi5 implementation

### DISCUSSION THREAD

**Discussion by ypatriot** [Joshua-Riek/ubuntu-rockchip#151] [General]
Dear All,
Every time I power up my orange pi 5 without internet connection, Time and date will be as the last time when I shutdown the orange pi5, it's not a current date and time.
I wonder can not find a clear way to use internal RTC on Orange Pi5, I need help to implement RTC internal or external such as RTC DS3231.
The DS3231 RTC was implemented on raspberry pi 4 succesfuly and it seem it will be able to be implemented on orange pi 5 because the GPIO pins 1,3,5,9 on the orange pi 5 are same function as raspberry pi 4. But I can not find the DS3231 driver on orange pi 5.

Is here some one please to help and give a guide if RTC DS3231 or may be another RTC module could be implemented on orange pi 5. Thank you very much in advance.
image: [the raspi4 RTC](https://www.arduinoecia.com.br/wp-content/uploads/2016/04/Real-Time-Clock-DS3231-Raspberry-Pi.png)

**Comment by HSTe** [comment]
I think you could find some information here:
[https://forum.armbian.com/topic/25674-orange-pi-5-opi5-rtc-real-time-clock/](url)

**Reply by ypatriot** [reply]
Thanks, 
Anyway I've read it many times, the solution is to connect RTC1-GND1 solder point with a 3.3V battery (CR2032)  on the bottom side board, that is a very small point to solder and I'm not sure to do this. 
is there another solution instead of soldering RTC1-GND1 to a battery,  may be using RTC module such as DS3231 or any other RTC module ?
